packages/react-aria-components/docs/styling.mdx

```diff
@@ -288,46 +288,61 @@ This enables using props like [animate](https://www.framer.com/motion/animation/
+Overlay exit animations can be implemented using the `isExiting` prop, which keeps the element in the DOM until an animation is complete. Framer Motion's [variants](https://www.framer.com/motion/animation/#variants) are a good way to setup named animation states.
 
 ```tsx
-import {AnimatePresence} from 'framer-motion';
+type AnimationState = 'unmounted' | 'hidden' | 'visible';
 
 function Example() {
-  let [isOpen, setOpen] = React.useState(false);
+  /*- begin highlight -*/
+  // Track animation state.
+  let [animation, setAnimation] = React.useState<AnimationState>('unmounted');
+  /*- end highlight -*/
 
   return (
-    <AnimatePresence>
-      {isOpen &&
-        <MotionModalOverlay
-          // Force mount so that AnimatePresence works.
-          /*- begin highlight -*/
-          isOpen
-          /*- end highlight -*/
-          onOpenChange={setOpen}
-          initial={{opacity: 0}}
-          animate={{opacity: 1}}
-          exit={{opacity: 0}}>
-          <MotionModal
-            initial={{opacity: 0, y: 32}}
-            animate={{opacity: 1, y: 0}}
-            exit={{opacity: 0, y: 32}}>
-            {/* ... */}
-          </MotionModal>
-        </MotionModalOverlay>
-      }
-    </AnimatePresence>
+    <DialogTrigger
+      /*- begin highlight -*/
+      // Start animation when open state changes.
+      onOpenChange={isOpen => setAnimation(isOpen ? 'visible' : 'hidden')}
+      /*- end highlight -*/
+    >
+      <Button>Open dialog</Button>
+      <MotionModalOverlay
+        /*- begin highlight -*/
+        // Prevent modal from unmounting during animation.
+        isExiting={animation === 'hidden'}
+        // Reset animation state once it is complete.
+        onAnimationComplete={animation => {
+          setAnimation(a => animation === 'hidden' && a === 'hidden' ? 'unmounted' : a)
+        }}
+        /*- end highlight -*/
+        variants={{
+          hidden: {opacity: 0},
+          visible: {opacity: 1}
+        }}
+        initial="hidden"
+        animate={animation}>
+        <MotionModal
+          variants={{
+            hidden: {opacity: 0, y: 32},
+            visible: {opacity: 1, y: 0}
+          }}>
+          {/* ... */}
+        </MotionModal>
+      </MotionModalOverlay>
+    </DialogTrigger>
   );
 }
 ```
 
-`AnimatePresence` also allows you to animate when items are added or removed in collection components. Use `array.map` to create children, and make sure each child has a unique `key` in addition to an `id` to ensure Framer Motion can track it.
+**Note**: Framer Motion's `AnimatePresence` component may not work with React Aria overlays in all cases, so the example shown above is the recommended approach for exit animations.
```

Comment:
Wasn't really sure where to put this. I thought a note about why we aren't using AnimatePresence might be important though? Sorta weird that we say this and then immediately start talking about it for collections below though.

Reply:
Yeah the flow here does feel a bit odd, but I think its fine since it serves as a forewarning